vpr/src/timing/timing_graph_builder.cpp

```diff
@@ -273,56 +518,56 @@ void TimingGraphBuilder::add_block_to_timing_graph(const AtomBlockId blk) {
         //and between the internal SOURCEs and SINKS for sequential blocks
         //
         //We have already marked all internal SOURCE/SINK nodes internal, and all IPIN/OPIN nodes as internal
-        //so we only need to look at tnode's of that class
+        //so we only need to look at tnode's of 'internal' class
         NodeId src_tnode = netlist_lookup_.atom_pin_tnode(src_pin, BlockTnode::INTERNAL);
 
-        if (src_tnode) {
-            auto src_type = tg_->node_type(src_tnode);
+        if (!src_tnode) continue;
 
-            //Look-up the combinationally connected sink ports name on the port model
-            AtomPortId src_port = netlist_.pin_port(src_pin);
-            const t_model_ports* model_port = netlist_.port_model(src_port);
+        auto src_type = tg_->node_type(src_tnode);
 
-            for (const std::string& sink_port_name : model_port->combinational_sink_ports) {
-                AtomPortId sink_port = netlist_.find_port(blk, sink_port_name);
-                if (!sink_port) continue; //Port may not be connected
+        //Look-up the combinationally connected sink ports name on the port model
+        AtomPortId src_port = netlist_.pin_port(src_pin);
+        const t_model_ports* model_port = netlist_.port_model(src_port);
 
-                //We now need to create edges between the source pin, and all the pins in the
-                //output port
-                for (AtomPinId sink_pin : netlist_.port_pins(sink_port)) {
-                    //Get the tnode of the sink
-                    NodeId sink_tnode = netlist_lookup_.atom_pin_tnode(sink_pin, BlockTnode::INTERNAL);
+        for (const std::string& sink_port_name : model_port->combinational_sink_ports) {
```

Comment:
This nesting is pretty deep.  Does it make sense to break down functionality into a named function?  Maybe "find_primative_connections_from_IPIN" or something?